### DISCUSSION TITLE
Possible InputStream leak

### DISCUSSION THREAD

**Discussion by mariantirlea** [belaban/JGroups#741] [Q&A] [answered]
I think I found an InputStream which is not closed and this can block the file after the JChannel object is destroyed.
I have an OSGI host and my module is closed but the udp.xml file remains blocked. After a few checks I noticed that there is an XmlConfiguration object created out of the stream but the stream is never closed.

Can anyone confirm this issue? I can help with the fix if needed...

```
    protected static XmlConfigurator parse(InputStream stream) throws java.io.IOException {
        try {
            XmlNode root=parseXmlDocument(stream);
            return parse(root);
        }
        catch (Exception x) {
            throw new IOException(String.format(Util.getMessage("ParseError"), x.getLocalizedMessage()));
        }
    }

    public static XmlConfigurator getInstance(InputStream stream) throws java.io.IOException {
        return parse(stream);
    }
```

**Comment by belaban** [comment]
This is definitely a bug. I guess a simple try()-finally clause wrapping the input stream would fix this...

**Comment by belaban** [comment]
https://issues.redhat.com/browse/JGRP-2731

**Comment by belaban** [comment, marked as answer]
OK, should be closed now. The commit is on `master` and `5.2.x` branches.

**Reply by mariantirlea** [reply]
Thank you for your fast response. It works now!

**Comment by belaban** [comment]
Perfect!
